source/deimos/openssl/err.d

```diff
@@ -327,7 +312,50 @@ struct ERR_string_data_st {
 	}
 alias ERR_string_data_st ERR_STRING_DATA;
 
-void ERR_put_error(int lib, int func,int reason,const(char)* file,int line);
+static if (OPENSSL_VERSION_BEFORE(3, 0, 0))
+{
+	void ERR_put_error(int lib, int func,int reason,const(char)* file,int line);
+	alias ERR_PUT_error = ERR_put_error;
+}
+else
+{
+	// New Error API
+
+	// 12 lines and some on an 80 column terminal
+	enum ERR_MAX_DATA_SIZE = 1024;
+
+	/* Building blocks */
+	void ERR_new();
+	void ERR_set_debug(const(char)* file, int line, const(char)* func);
+	void ERR_set_error(int lib, int reason, const(char)* fmt, ...);
+	void ERR_vset_error(int lib, int reason, const(char)* fmt, va_list args);
+
+	auto ERR_raise () (int lib, int reason) {
+		return ERR_raise_data(lib, reason, null);
+	}
+
+	// WARNING: The string passed to this function must be `\0` terminated
+	// This works for literal, but for non-default arguments, beware.
+	// Note that the macro in OpenSSL v3 does not take argument,
+	// but we need to if we want to present the same API.
+	auto ERR_raise_data () (
+		string file = __FILE__, int line = __LINE__, string func = __FUNCTION__)
```

Comment:
Signature in [docs](https://www.openssl.org/docs/manmaster/man3/ERR_raise_data.html) seems to be:

    void ERR_raise_data(int lib, int reason, const char *fmt, ...);